File: 3606-minimum-element-after-replacement-with-digit-sum/minimum-element-after-replacement-with-digit-sum.swift
class Solution {
    func minElement(_ nums: [Int]) -> Int {
        var strArr: [String] = nums.map{String($0)}
        var resArr: [Int] = []
        var sum = 0

        for i in strArr{
            sum = i.compactMap(\.wholeNumberValue).reduce(0, +)
            resArr.append(sum)
        }

        return resArr.min() ?? 0

        
    }
}

File: 3606-minimum-element-after-replacement-with-digit-sum/minimum-element-after-replacement-with-digit-sum.py
class Solution:
    def minElement(self, nums: List[int]) -> int:
        arr = []
        for num in nums:
            sum = 0
            for i in str(num):
                sum += int(i)
            arr.append(sum)

        return min(arr)


        
            
         
        
        